### PR TITLE
SH-5972: Use default Drupal layouts in Dashboard

### PR DESCRIPTION
**Pull request by mariannuar**
# READY FOR REVIEW

## Summary
Update logic in hs_layouts.module to use default Drupal layouts in Dashboards instead of the onces used in the FE

## Urgency
medium

## Steps to Test
1. Go to https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com/admin/structure/dashboard/experimental_dashboard/preview
2. Edit layout
3. Add a section and confirm there are only four layouts (1 Col, 2 Cols, 3 Cols, 4 Cols) that can be used in Dasboard
4. Add any of them and confirm they look good (there will another ticket to make it looks better, once there's a design)
5. Go to any page in the site (for example /accordions/accordions-secondary)
6. Confirm the same custom layouts can be used and not the default by Drupal

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208840953610430

### CODE REVIEW

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-colorful-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-traditional-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://english-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://history-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://west-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852

Visual Diffs:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-colorful-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-traditional-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://english-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://history-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://west-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852

Visual Diffs:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852/visualdiffs

File: docroot/modules/humsci/hs_layouts/hs_layouts.module [hunk 1]
```diff
@@ -13,8 +13,16 @@ use Drupal\paragraphs\ParagraphInterface;
  */
 function hs_layouts_plugin_filter_layout__layout_builder_alter(&$definitions, &$extra, &$consumer) {
   /** @var \Drupal\Core\Layout\LayoutDefinition $definition */
+  $active_theme = \Drupal::theme()->getActiveTheme()->getName();
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.module]: I was checking and there's a more robust way to test if we're in the Dashboard. `$extra` includes a `section_storage` element that will be an instance of `Drupal\dashboard\Plugin\SectionStorage\DashboardSectionStorage` if the current layout builder instance is a dashboard.

Also, having all conditions in a single `if` is a bit confusing. I'd change it to something like this:

```
if ($extra['section_storage'] instanceof DashboardSectionStorage) {
  // Logic for filtering the sections that will be available for the dashboard.
}
else {
  // Logic for keeping only the sections defined in the `hs_layout` module.
}
```

Also:
- Make sure to add `use Drupal\dashboard\Plugin\SectionStorage\DashboardSectionStorage;` in the top of the file.
- Add the dashboard module as a dependency for `hs_layouts` in the `info.yml` file

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar That's a good start, just added some notes for doing the implementation more robust and clear.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-colorful-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-traditional-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://english-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://history-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://west-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852

Visual Diffs:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852/visualdiffs

File: docroot/modules/humsci/hs_layouts/hs_layouts.info.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ version: 8.2.7
 package: 'Humanities & Sciences'
 dependencies:
   - layout_discovery
+  - drupal/dashboard
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.info.yml]: This is not the right format for dependencies. See https://www.drupal.org/docs/develop/theming-drupal/defining-a-theme-with-an-infoyml-file#dependencies

```suggestion
  - dashboard:dashboard
```

File: docroot/modules/humsci/hs_layouts/hs_layouts.module [hunk 3]
```diff
@@ -7,15 +7,29 @@
 
 use Drupal\Component\Utility\Html;
 use Drupal\paragraphs\ParagraphInterface;
+use Drupal\dashboard\Plugin\SectionStorage\DashboardSectionStorage;
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.module:9]: `use` statements should be sorted alphabetically.

File: docroot/modules/humsci/hs_layouts/hs_layouts.module [hunk 4]
```diff
@@ -7,15 +7,29 @@
...
  * Implements hook_plugin_filter_TYPE__CONSUMER_alter().
  */
 function hs_layouts_plugin_filter_layout__layout_builder_alter(&$definitions, &$extra, &$consumer) {
   /** @var \Drupal\Core\Layout\LayoutDefinition $definition */
   foreach ($definitions as $id => $definition) {
-    if ($definition->getProvider() !== 'hs_layouts') {
-      unset($definitions[$id]);
+    $provider = $definition->getProvider();
+    if ($extra['section_storage'] instanceof DashboardSectionStorage) {
+      // Define providers to unset for the gin admin theme.
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.module]:
```suggestion
      // Define providers to unset for the dashboard.
```

File: docroot/modules/humsci/hs_layouts/hs_layouts.module [hunk 5]
```diff
@@ -7,15 +7,29 @@
...
+      $admin_providers_to_unset = [
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.module]: Try to keep variable names under 20 char, otherwise we'll get a CodeClimate error.

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Almost there, just a couple of things to fix.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-colorful-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-traditional-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://english-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://history-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://west-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852

Visual Diffs:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852/visualdiffs

**Comment by mariannuar** [comment]
@cienvaras Done!

File: docroot/modules/humsci/hs_layouts/hs_layouts.info.yml (hunk 2 above)
Comment (mariannuar) [reply docroot/modules/humsci/hs_layouts/hs_layouts.info.yml]: Oh 👀 thanks!

File: docroot/modules/humsci/hs_layouts/hs_layouts.info.yml [hunk 6]
```diff
@@ -6,6 +6,7 @@ version: 8.2.7
 package: 'Humanities & Sciences'
 dependencies:
   - layout_discovery
+  - drupal:dashboard
```
Comment (cienvaras) [inline docroot/modules/humsci/hs_layouts/hs_layouts.info.yml]: For contrib modules you should add the module name twice. `drupal:something` is only for core modules.

```suggestion
  - dashboard:dashboard
```

**Review by cienvaras — CHANGES_REQUESTED** [review]
@mariannuar Only one change needed.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1697-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-colorful-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://hs-traditional-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://english-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://history-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com
- https://west-pksvk1y2mlhnkwglqkecukewc9y0oft0.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852

Visual Diffs:
- https://dashboard.tugboatqa.com/675b6209252c7a7b92288852/visualdiffs

**Review by cienvaras — APPROVED** [review]
@mariannuar Thanks, looks good. Merging into the longterm dashboard branch.

**Pull request merged**